### PR TITLE
Fix where we got double nullable for `nullable` types

### PR DESCRIPTION
**Pull request by mcjonesy-nz**
When something was nullable, generated two ??

### CODE REVIEW

File: src/SigSpec.CodeGeneration.CSharp.Tests/NullableGeneratorTests.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using SigSpec.Core;
+using Xunit;
+
+namespace SigSpec.CodeGeneration.CSharp.Tests;
+
+public class NullableGeneratorTests
+{
+	private readonly SigSpecGeneratorSettings _settings = new SigSpecGeneratorSettings();
```
Comment (danzel) [inline src/SigSpec.CodeGeneration.CSharp.Tests/NullableGeneratorTests.cs:11]: formatting a bit wacky in here

File: src/SigSpec.Console/Program.cs [hunk 2]
```diff
@@ -32,17 +32,24 @@ static async Task RunAsync()
...
-            var tsCodeGeneratorSettings = new SigSpecToTypeScriptGeneratorSettings();
-            var tsCodeGenerator = new SigSpecToTypeScriptGenerator(tsCodeGeneratorSettings);
-            var file = tsCodeGenerator.GenerateFile(document);
-
-            Console.WriteLine("\n\nGenerated SigSpec TypeScript code:");
-            Console.WriteLine(file);
-            Console.ReadKey();
-
-            var codeGeneratorSettings = new SigSpecToCSharpGeneratorSettings();
+            // Typescript generation broken due to njsonschema.codegeneration update
```
Comment (danzel) [inline src/SigSpec.Console/Program.cs]: Need to fix this, I think updating all references will fix it?

File: src/SigSpec.CodeGeneration.CSharp.Tests/GeneratorTests.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		{
+			_generator = new SigSpecGenerator(_settings);
+			_codeGenerator = new SigSpecToCSharpGenerator(_codeGeneratorSettings);
+		}
+
+		[Theory]
+		[InlineData(typeof(HubWithString), "string")]
+		[InlineData(typeof(HubWithStringOptional), "string")]
+		[InlineData(typeof(HubWithInt), "int")]
+		[InlineData(typeof(HubWithIntOptional), "int")]
```
Comment (danzel) [inline src/SigSpec.CodeGeneration.CSharp.Tests/GeneratorTests.cs]: Value types should still get a ? in this test I think (or be `Nullable<T>`).
(int, Guid, Struct)
They are value types, so `GenerateNullableReferenceTypes = false` shouldn't stop them.

**Review by danzel — CHANGES_REQUESTED** [review]

File: src/SigSpec.CodeGeneration.CSharp.Tests/GeneratorTests.cs (hunk 3 above)
Comment (mcjonesy-nz) [reply src/SigSpec.CodeGeneration.CSharp.Tests/GeneratorTests.cs]: I'm not sure about having that inconsistency then? 
Perhaps re-add our own options flag and use that for generating nullable or not.

Comment (danzel) [reply src/SigSpec.CodeGeneration.CSharp.Tests/GeneratorTests.cs]: As discussed IRL (for anyone else following along).
Nullable value types are an old c# feature, nullable reference types are a new c# feature.
The `GenerateNullableReferenceTypes = false` option is for older c# back compatibility.

File: src/SigSpec.CodeGeneration.CSharp.Tests/GeneratorTests.cs [hunk 4]
```diff
@@ -28,15 +31,14 @@ public GeneratorTests()
...
 		[InlineData(typeof(HubWithStringOptional), "string")]
 		[InlineData(typeof(HubWithInt), "int")]
-		[InlineData(typeof(HubWithIntOptional), "int")]
+		[InlineData(typeof(HubWithIntOptional), "int?")]
 		[InlineData(typeof(HubWithGuid), "System.Guid")]
-		[InlineData(typeof(HubWithGuidOptional), "System.Guid")]
+		[InlineData(typeof(HubWithGuidOptional), "System.Guid?")]
 		[InlineData(typeof(HubWithStruct), "TestStruct")]
 		[InlineData(typeof(HubWithNullableStruct), "TestStruct")]
 		[InlineData(typeof(HubWithStructOptional), "TestStruct")]
```
Comment (danzel) [inline src/SigSpec.CodeGeneration.CSharp.Tests/GeneratorTests.cs]: I believe HubWithNullableStruct and HubWithStructOptional should both be `?` or `Nullable<T>` here.

**Pull request merged**